test/test_nn.py

```diff
@@ -5434,6 +5434,38 @@ def test_pad_scalar_error(self):
         self.assertRaises(RuntimeError, lambda: F.pad(inputs, (1, 1)))
         self.assertRaises(RuntimeError, lambda: F.pad(inputs, (1,)))
 
+    def test_nested_tensor_from_mask(self):
```

Comment:
Could possibly include tests validating that errors are thrown when shapes are mismatched